### PR TITLE
Added basic config stuff

### PR DESCRIPTION
**Pull request by fireteam99**
# Description

Adds support for game configs.
Please merge with: https://github.com/chinese-board-games/luzhanqi-web/pull/105

## Type of Change

- [x] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Documentation update

## Checklist

- [ ] Unit/integration tests
- [ ] Documentation

### CODE REVIEW

**Review by iankhou — CHANGES_REQUESTED** [review]

File: src/lzqgame.ts [hunk 1]
```diff
@@ -619,7 +656,11 @@ async function playerForfeit(
     const playerIndex = myGame.players.indexOf(playerName);
     const winnerIndex = playerIndex === 0 ? 1 : 0;
 
-    await updateGame(gid, { winnerId: myGame.playerToUidMap.get(myGame.players[winnerIndex]) || 'anonymous' });
+    await updateGame(gid, {
+        winnerId:
+            myGame.playerToUidMap.get(myGame.players[winnerIndex]) ||
+            'anonymous',
+    });
```
[the remark this answers is not in the document]
Comment (iankhou) [reply src/lzqgame.ts:663]: Nevermind, it was my own linter issue.

[the remark this answers is not in the document]
Comment (fireteam99) [reply src/lzqgame.ts:663]: addressed async

File: src/models/Game.ts [hunk 2]
```diff
@@ -1,4 +1,11 @@
-import { Schema, model } from 'mongoose';
+import { Schema, model, Types, Model } from 'mongoose';
+
+interface IGameConfig extends Document {
+    _id: Types.ObjectId;
```
[the remark this answers is not in the document]
Comment (fireteam99) [reply src/models/Game.ts:4]: Yes, it's required for subdocuments: https://mongoosejs.com/docs/typescript/subdocuments.html

File: package.json [hunk 3]
```diff
@@ -33,7 +33,7 @@
         "lodash": "^4.17.21",
         "lodash.isequal": "^4.5.0",
         "mongodb": "^3.6.3",
-        "mongoose": "^5.11.8",
+        "mongoose": "^8.0.0",
```
[the remark this answers is not in the document]
Comment (fireteam99) [reply package.json:36]: Upgrading to 8 was the only way to get subdocuments to work with TS ([docs](https://mongoosejs.com/docs/typescript/subdocuments.html)). There don't appear to be any regressions so we should be ok

File: src/lzqgame.ts [hunk 4]
```diff
@@ -90,14 +100,26 @@ async function hostCreateNewGame(
  * Two players have joined. Alert the host!
  * @param roomId The room ID
  */
-function hostPrepareGame(this: Socket, gid: string) {
+async function hostPrepareGame(
+    this: Socket,
+    roomId: string,
```
Comment (iankhou) [inline src/lzqgame.ts:105]: roomId and gameId are both strings of the same value (stringified Mongo ObjectId). Should be only one param that represents both (roomId on client-side, gid on server-side).

**Review by iankhou — CHANGES_REQUESTED** [review]

**Review by iankhou — APPROVED** [review]
Nice

**Pull request merged**